### PR TITLE
V1.3.2 track config improvements

### PR DESCRIPTION
**Pull request by brentpicasso**
manual track config customization improvements. 

* allow updating of target from gps point
* streamlined text fields
* point customization dialog

### CODE REVIEW

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.kv [hunk 1]
```diff
@@ -6,12 +6,72 @@
...
+                    id: lat
+                    on_text: root.on_latitude(*args)                
+                FloatValueField:
+                    id: lon
+                    on_text: root.on_longitude(*args)
+            BoxLayout:
+                size_hint_y: 0.4
+                
+        BoxLayout:
+            #size_hint_x: 0.2
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.kv]: Probably not needed?

Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.kv]: ah yes.

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.kv [hunk 2]
```diff
@@ -6,12 +6,72 @@
...
+            orientation: 'vertical'
+            spacing: dp(20)
+            padding: (dp(5), dp(5))
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.kv]: dp() is the preferred/better method to handle different pixel densities, correct?

Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.kv]: typically that's what they recommend. Although, much of the scaling stuff we've been doing for handling phone screen sizes involves just scaling it to the window or current view's height

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py [hunk 3]
```diff
@@ -17,49 +19,143 @@
...
+        self.databus = kwargs.get('databus')
         self.register_event_type('on_config_changed')
         title = kwargs.get('title', None)
         if title:
             self.setTitle(title)
+        Clock.schedule_interval(lambda dt: self.update_gps_status(), GPS_STATUS_POLL_INTERVAL)
 
+    def _get_gps_quality(self):
+        gps_quality = self.databus.channel_data.get('GPSQual')
+        return 0 if gps_quality is None else int(gps_quality)
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: What's the behavior if the GPSQual channel is disabled?

Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: it returns None and should be safely handled by line 44

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py [hunk 4]
```diff
@@ -17,49 +19,143 @@
...
-        self.lonView.text = str(geoPoint.longitude)
-        self.latView.set_next(self.lonView)        
-        self.geoPoint = geoPoint
+    def on_update_target(self, *args):
+        try:
+            if self._get_gps_quality() >= GpsConfig.GPS_QUALITY_2D:
+                channel_data = self.databus.channel_data
+                point = self.point
+                point.latitude = channel_data.get('Latitude')
+                point.longitude = channel_data.get('Longitude')
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: Do we have to worry about timing issues where GPS is locked but due to the update rate the lat/long channels may not be populated?

Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: in a practical sense, no.

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py [hunk 5]
```diff
@@ -17,49 +19,143 @@
...
+                          size_hint=(None, None), size = (dp(500), dp(220)))
+            content.bind(on_close=lambda *args:popup.dismiss())                     
+            content.bind(on_point_edited=self._on_edited)
+            popup.open()
+        
+    def _refresh_point_view(self):
+        self.ids.lat.text = str(self.point.latitude)
+        self.ids.lon.text = str(self.point.longitude)
+        
+    def setPoint(self, point):
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: Not PEP8 compliant :)

Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: (╯°□°）╯︵ ┻━┻

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py [hunk 6]
```diff
@@ -17,49 +19,143 @@
...
+        self.point = point
+        self._refresh_point_view()
+            
+class GeoPointEditor(BoxLayout):
+    def __init__(self, **kwargs):
+        super(GeoPointEditor, self).__init__(**kwargs)
+        self.point = kwargs.get('point')
+        self.databus = kwargs.get('databus')
+        self.register_event_type('on_point_edited')
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: This reminds me of my comments a long time about about kwargs vs regular arguments. http://docs.python-guide.org/en/latest/writing/style/#function-arguments says keyword arguments are not mandatory. So if we have mandatory arguments we should define them. Thoughts?

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.kv (hunk 2 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.kv]: :+1:

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py (hunk 3 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: :+1:

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py (hunk 6 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: you mean having it create GeoPointEditor as  GeoPointEditor(point, databus)?

I think that would be a fine practice. I checked, and you can still do keyword arguments for positional mandatory arguments, if one feels that makes the code more readable.

Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: Yeah, that's what I was thinking. It makes it very easy to see what a class or function requires and allows for automatic tools to introspect code.

**Comment by ryandoherty** [comment]
Crash, STR:
1) Click the gear icon for the start/finish
2) Paste via CMD/CTRL-V into lat point: ', -122.057135'

**Comment by ryandoherty** [comment]
Overall looks good! Just those few comments and one crash bug.

**Comment by brentpicasso** [comment]
fixed PEP8, crashing and required params on GeoPointEditor

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py (hunk 5 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: fixed

File: autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py (hunk 6 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/trackconfigview.py]: updated.

**Pull request merged**